sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -958,18 +958,25 @@ object QueryExecutionErrors {
     new RuntimeException("Parsing JSON arrays as structs is forbidden.")
   }
 
-  def cannotParseStringAsDataTypeError(str: String, dataType: DataType): Throwable = {
-    new RuntimeException(s"Cannot parse $str as ${dataType.catalogString}.")
+  def cannotParseStringAsDataTypeError(parser: JsonParser, token: JsonToken, dataType: DataType)
+  : Throwable = {
+    new RuntimeException(
+      s"Cannot parse field name [${parser.getCurrentName}], " +
+        s"field value [${parser.getText}], " +
```

Comment:
Any reason to output field name and values in such form? Could you check other places in Spark code base, and look for other forms of field names and values in error messages.